File: ts/wam/6412.ts
export const WamThreadInteractionDataNotification = 6412 as const

export interface ThreadInteractionDataNotificationEvent {
  /** field 1, wire `group_size_bucket` */
  groupSizeBucket?: CLIENTGROUPSIZEBUCKET
  /** field 2, wire `group_type_client` */
  groupTypeClient?: GROUPTYPECLIENT
  /** field 3, wire `is_a_group` */
  isAGroup?: boolean
  /** field 4, wire `thread_ds` */
  threadDs?: string
  /** field 6, wire `total_link_reshare_message_notif_shown` */
  totalLinkReshareMessageNotifShown?: number
  /** field 7, wire `total_link_reshare_message_notif_shown_fb` */
  totalLinkReshareMessageNotifShownFb?: number
  /** field 8, wire `total_link_reshare_message_notif_shown_ig` */
  totalLinkReshareMessageNotifShownIg?: number
  /** field 9, wire `total_link_reshare_message_notif_tap_to_open` */
  totalLinkReshareMessageNotifTapToOpen?: number
  /** field 10, wire `total_link_reshare_message_notif_tap_to_open_fb` */
  totalLinkReshareMessageNotifTapToOpenFb?: number
  /** field 11, wire `total_link_reshare_message_notif_tap_to_open_ig` */
  totalLinkReshareMessageNotifTapToOpenIg?: number
  /** field 12, wire `total_message_reminder_notif_shown` */
  totalMessageReminderNotifShown?: number
  /** field 13, wire `total_message_reminder_notif_tap_to_open` */
  totalMessageReminderNotifTapToOpen?: number
  /** field 14, wire `total_notif_mark_as_read` */
  totalNotifMarkAsRead?: number
  /** field 15, wire `total_notif_missed_call_voip_callback` */
  totalNotifMissedCallVoipCallback?: number
  /** field 16, wire `total_notif_missed_call_voip_message` */
  totalNotifMissedCallVoipMessage?: number
  /** field 17, wire `total_notif_others` */
  totalNotifOthers?: number
  /** field 18, wire `total_notif_reply` */
  totalNotifReply?: number
  /** field 19, wire `total_notif_rtc_voip_accept` */
  totalNotifRtcVoipAccept?: number
  /** field 20, wire `total_notif_rtc_voip_decline` */
  totalNotifRtcVoipDecline?: number
  /** field 21, wire `total_notif_show_preview` */
  totalNotifShowPreview?: number
  /** field 22, wire `total_notif_shown` */
  totalNotifShown?: number
  /** field 23, wire `total_notif_tap_to_open` */
  totalNotifTapToOpen?: number
  /** field 24, wire `thread_id` */
  threadId?: string
  /** field 25, wire `thread_id_by_lid` */
  threadIdByLid?: string
  /** field 26, wire `thread_creation_date` */
  threadCreationDate?: string
}

File: go/wam/6412.go
const WamThreadInteractionDataNotification = 6412 // channel: regular

type ThreadInteractionDataNotificationEvent struct {
	GroupSizeBucket CLIENTGROUPSIZEBUCKET `wam:"group_size_bucket"` // field 1
	GroupTypeClient GROUPTYPECLIENT `wam:"group_type_client"` // field 2
	IsAGroup bool `wam:"is_a_group"` // field 3
	ThreadDs string `wam:"thread_ds"` // field 4
	TotalLinkReshareMessageNotifShown int64 `wam:"total_link_reshare_message_notif_shown"` // field 6
	TotalLinkReshareMessageNotifShownFb int64 `wam:"total_link_reshare_message_notif_shown_fb"` // field 7
	TotalLinkReshareMessageNotifShownIg int64 `wam:"total_link_reshare_message_notif_shown_ig"` // field 8
	TotalLinkReshareMessageNotifTapToOpen int64 `wam:"total_link_reshare_message_notif_tap_to_open"` // field 9
	TotalLinkReshareMessageNotifTapToOpenFb int64 `wam:"total_link_reshare_message_notif_tap_to_open_fb"` // field 10
	TotalLinkReshareMessageNotifTapToOpenIg int64 `wam:"total_link_reshare_message_notif_tap_to_open_ig"` // field 11
	TotalMessageReminderNotifShown int64 `wam:"total_message_reminder_notif_shown"` // field 12
	TotalMessageReminderNotifTapToOpen int64 `wam:"total_message_reminder_notif_tap_to_open"` // field 13
	TotalNotifMarkAsRead int64 `wam:"total_notif_mark_as_read"` // field 14
	TotalNotifMissedCallVoipCallback int64 `wam:"total_notif_missed_call_voip_callback"` // field 15
	TotalNotifMissedCallVoipMessage int64 `wam:"total_notif_missed_call_voip_message"` // field 16
	TotalNotifOthers int64 `wam:"total_notif_others"` // field 17
	TotalNotifReply int64 `wam:"total_notif_reply"` // field 18
	TotalNotifRtcVoipAccept int64 `wam:"total_notif_rtc_voip_accept"` // field 19
	TotalNotifRtcVoipDecline int64 `wam:"total_notif_rtc_voip_decline"` // field 20
	TotalNotifShowPreview int64 `wam:"total_notif_show_preview"` // field 21
	TotalNotifShown int64 `wam:"total_notif_shown"` // field 22
	TotalNotifTapToOpen int64 `wam:"total_notif_tap_to_open"` // field 23
	ThreadId string `wam:"thread_id"` // field 24
	ThreadIdByLid string `wam:"thread_id_by_lid"` // field 25
	ThreadCreationDate string `wam:"thread_creation_date"` // field 26
}